app/Modules/SessionRequest/SessionRequestService.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+namespace App\Modules\SessionRequest;
+
+use App\Modules\Bookings\Booking;
+use App\Modules\Bookings\Services\BookingsService;
+use App\Modules\Position\Position;
+use App\Modules\SessionRequest\Exceptions\SessionRequestAlreadyAcceptedException;
+use App\Modules\SessionRequest\Exceptions\SessionRequestAlreadyExistsException;
+use App\Modules\SessionRequest\Exceptions\SessionRequestNotAcceptedException;
+use App\User;
+use Carbon\Carbon;
+
+class SessionRequestService
+{
+    private BookingsService $bookingsService;
+
+    public function __construct(BookingsService $bookingsService)
+    {
+        $this->bookingsService = $bookingsService;
+    }
+
+    public function createSessionRequest(User $user, Position $position)
+    {
+        $existingSessionRequests = SessionRequest::where('user_id', $user->id)->where(
+            'position_id',
+            $position->id
+        )->get();
+
+        throw_if(! $existingSessionRequests->isEmpty(), SessionRequestAlreadyExistsException::class);
```

Comment:
Do we need to consider pilot vs atc session requests here?

Reply:
That concept doesn't yet exist, but I suspect we will in the future